src/hotspot/share/compiler/compileBroker.cpp

```diff
@@ -946,12 +946,13 @@ void CompileBroker::init_compiler_threads() {
     jobject thread_handle = nullptr;
     // Create all j.l.Thread objects for C1 and C2 threads here, but only one
     // for JVMCI compiler which can create further ones on demand.
-    JVMCI_ONLY(if (!UseJVMCICompiler || !UseDynamicNumberOfCompilerThreads || i == 0) {)
-    // Create a name for our thread.
-    os::snprintf_checked(name_buffer, sizeof(name_buffer), "%s CompilerThread%d", _compilers[1]->name(), i);
-    Handle thread_oop = create_thread_oop(name_buffer, CHECK);
-    thread_handle = JNIHandles::make_global(thread_oop);
-    JVMCI_ONLY(})
+    bool create = !UseDynamicNumberOfCompilerThreads || i == 0;
+    if (create JVMCI_ONLY(|| !UseJVMCICompiler || UseJVMCINativeLibrary)) {
```

Comment:
Why it is conditional for C2 now? It created always if UseJVMCICompiler is false before.
And if you want to match this for Graal you just need to remove the condition.